dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java

```diff
@@ -0,0 +1,239 @@
+package com.dotcms.rendering.velocity.servlet;
+
+import com.dotcms.api.web.HttpServletRequestThreadLocal;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.datagen.*;
+import com.dotcms.util.FiltersUtil;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Clickstream;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.filters.VanityURLFilter;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.WebKeys;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.dotcms.datagen.TestDataUtils.getNewsLikeContentType;
+
+import static org.mockito.Mockito.*;
+
+public class VelocityServletIntegrationTest {
+
+    public static final String TEST_PATTERN = "/testpattern";
+    private HttpServletRequest request;
+    private HttpServletResponse response;
+    private Host host;
+    private ServletOutputStream servletOutputStream;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    @Before
+    public void init() {
+        host = new SiteDataGen().nextPersisted();
+
+        final Map<String, Object> attributes = new HashMap<>();
+
+        request = mock(HttpServletRequest.class);
+        // Mock setAttribute
+        Mockito.doAnswer(new Answer<Void>() {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable {
+                String key = invocation.getArgumentAt(0, String.class);
+                Object value = invocation.getArgumentAt(1, Object.class);
+                attributes.put(key, value);
+                return null;
+            }
+        }).when(request).setAttribute(Mockito.anyString(), Mockito.anyObject());
+
+        // Mock getAttribute
+        Mockito.doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'invocation' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4842460)